app/controllers/articles_controller.rb

```diff
@@ -1,72 +1,64 @@
 class ArticlesController < ApplicationController
 
-before_action :authenticate_user!, only: [:destroy, :delete, :new, :create, :edit, :update]
+  before_action :authenticate_user!, only: [:destroy, :delete, :new, :create, :edit, :update]
+  def new
+    @article = Article.new
+  end
 
-	def new
-		@article = Article.new
-	end
+  def create
+    @user = User.find(current_user.id)
+    @article = @user.articles.build(article_params)
+    if @article.save
+      redirect_to article_path(@article), :notice => "Article successfully added"
+    else
+      render :new
+    end
+  end
 
-	def create
-		@user = User.find(current_user.id)
+  def index
+    @articles = Article.order(created_at: :desc).page(params[:page]).per(10)
+  end
 
-		@article = @user.articles.build(article_params)
-	
-		if @article.save
-			redirect_to article_path(@article), :notice => "Article successfully added"
-		else
-			render :new
-		end
-	end
-
-	def index
-		@articles = Article.order(created_at: :desc).page(params[:page]).per(10)
-	end
-
-	def show
-		
-		@article = Article.find(params[:id])
+  def show
+    @article = Article.find(params[:id])
     @reviews = @article.reviews.order(created_at: :desc)
-		@review = Review.new
-	end
-
-	def edit
-		@article = Article.find(params[:id])
-
-	end
+    @review = Review.new
+  end
 
-	def update
-		@article = Article.find(params[:id])
-		if current_user.id != @article.user_id
-			redirect_to @article, :notice => "Invalid user"
-		else
-			if @article.update_attributes(article_params)
-				redirect_to edit_article_path, :notice => "Article successfully edited"
-			else
-				render :edit, :notice => "Article did not update"
-			end
-		end
-	end
+  def edit
+    @article = Article.find(params[:id])
+  end
 
-	def destroy
-   @article = Article.find(params[:id])
-		if current_user.id != @article.user_id
+  def update
+    @article = Article.find(params[:id])
+    if current_user.id != @article.user_id
+      redirect_to @article, :notice => "Invalid user"
+    else
+      if @article.update_attributes(article_params)
+        redirect_to edit_article_path, :notice => "Article successfully edited"
+      else
+        render :edit, :notice => "Article did not update"
+      end
+    end
+  end
 
-			redirect_to @article, :notice => "Invalid user"
-		else
-			@article = Article.find(params[:id]).destroy
-			
-			if Article.exists?(@article) == false
-				
-				redirect_to articles_path, :notice => "Article successfully deleted"
-			else
-				render :edit, :notice => "Article did not delete"
-			end
-		end	
-	end
+  def destroy
+    @article = Article.find(params[:id])
+    if current_user.id != @article.user_id
+      redirect_to @article, :notice => "Invalid user"
+    else
+      @article = Article.find(params[:id]).destroy
+      if @article.destroy
+        redirect_to articles_path, :notice => "Article successfully deleted"
+      else
+        render :edit, :notice => "Article did not delete"
```

Comment:
Don't think you can call a flash after a render since it's not issuing another request.  Normally on a render if you want a flash you need to use flash.now.notice = "whatever".  Try getting rid of this and see if it fixes the build